### DISCUSSION TITLE
How to access Vault in the root namespace?

### DISCUSSION THREAD

**Discussion by pseudoscion** [rajanadar/VaultSharp#360] [Q&A]
I have a client who has a HashiCorp Vault, with their secrets within the root/empty namespace. This seems to be causing problem when trying to access the secrets. Unfortunately, the test vault that I have access to restricts me from creating secrets engines within the root namespace, thus I am unable to test/debug this properly. 
I am creating my VaultClient as so,
 ```
  VaultClientSettings vaultSettings = new VaultClientSettings(vURI, vAuthMethod);

  if (GetConfigSetting(APPSETTINGS_VAULT_NAMESPACE, out string vNamespace, OPTIONAL) && !String.IsNullOrWhiteSpace(vNamespace))
  {
    vaultSettings.Namespace = vNamespace;
  }

  fVaultClient = new VaultClient(vaultSettings);
```
When using the default namespace, do I need to specify anything special within the VaultClient object?

Any help would be greatly appreciated.

**Comment by rajanadar** [comment]
What you are doing looks okay. For the default namespace, you don't need to specify anything special.
